validator/client/grpc-api/grpc_validator_client.go

```diff
@@ -302,6 +344,46 @@ func (c *grpcValidatorClient) StartEventStream(ctx context.Context, topics []str
 	}
 }
 
+func (c *grpcValidatorClient) StartEventStream(ctx context.Context, topics []string, eventsChannel chan<- *eventClient.Event) {
+	ctx, span := trace.StartSpan(ctx, "validator.gRPCClient.StartEventStream")
+	defer span.End()
+	if len(topics) == 0 {
+		eventsChannel <- &eventClient.Event{
+			EventType: eventClient.EventError,
+			Data:      []byte(errors.New("no topics were added").Error()),
+		}
+		return
+	}
+	containsUnsupported := false
+	var supportedRequested []string
+	for _, topic := range topics {
+		_, supported := supportedEvents[topic]
+		if supported {
+			supportedRequested = append(supportedRequested, topic)
+		} else {
+			containsUnsupported = true
+		}
+	}
+	if containsUnsupported {
+		log.Warn("Requested unsuported event topics, they will be ignored")
+	}
+	if len(supportedRequested) == 0 {
+		eventsChannel <- &eventClient.Event{
+			EventType: eventClient.EventError,
+			Data:      []byte(errors.New("no topics were added").Error()),
+		}
+		return
+	}
+	for _, topic := range supportedRequested {
+		switch topic {
+		case eventClient.EventHead:
+			go c.streamHead(ctx, eventsChannel)
```

Comment:
you might need to use a sync.WaitGroup for closing this channel instead of doing it inside